mtags/src/main/scala/scala/meta/internal/pc/Completions.scala

```diff
@@ -1632,6 +1675,93 @@ trait Completions { this: MetalsGlobal =>
         }
       }
     }
+
+    /**
+     * A scaladoc completion showing the parameters of the given associated definition.
+     *
+     * @param editRange the range in the original source file.
+     * @param associatedDef the memberDef associated with the scaladoc to complete.
+     *                      This class will construct scaladoc based on the params of this definition.
+     */
+    case class Scaladoc(
+        editRange: l.Range,
+        associatedDef: MemberDef
+    ) extends CompletionPosition {
+      override def contribute: List[Member] = {
+        val lines = scaladocLines(associatedDef)
+        // add `* @return` line only if the definition is a method def.
+        val returnLine =
+          if (lines.isEmpty && clientSupportsSnippets) "* @return $0"
+          else "* @return"
+        val newText =
+          if (associatedDef.isInstanceOf[DefDef])
+            (Seq("", "*") ++ lines ++ Seq(returnLine, "*/")).mkString("\n  ")
+          else (Seq("", "*") ++ lines ++ Seq("*/")).mkString("\n  ")
+        List(
+          new TextEditMember(
+            "Scaladoc Comment",
+            new l.TextEdit(
+              editRange,
+              newText
+            ),
+            completionsSymbol(associatedDef.name.toString()),
+            label = Some("/** */"),
+            detail = Some("Scaladoc Comment")
+          )
+        )
+      }
+    }
+
+    /**
+     * Returns the parameter lines of scaladoc besed on the given memberDef
+     * like ["* @param param1 $0", "* @param param2"].
+     *
+     * @param memberDef The memberDef to construct scaladoc.
+     */
+    private def scaladocLines(memberDef: MemberDef): List[String] = {
+      memberDef match {
+        case DefDef(_, _, _, vparamss, _, _) =>
```

Comment:
```suggestion
        case d: DefDef =>
```

and then `d.vparamss`